src/Frontend/Page/Admin.hs

```diff
@@ -342,15 +402,36 @@ adminSettingsGaPUsersCreate =
 
 adminSettingsGaPClassesView :: ActionM r m => m (Frame PageAdminSettingsGaPClassesView)
 adminSettingsGaPClassesView =
-    makeFrame =<< PageAdminSettingsGaPClassesView . mapMaybe toClass <$> persistent getSpaces
-  where
-    toClass (ClassSpace clss) = Just clss
-    toClass _                 = Nothing
+    makeFrame =<< PageAdminSettingsGaPClassesView <$> persistent getSchoolClasses
 
 adminSettingsGaPClassesCreate :: ActionM r m => m (Frame PageAdminSettingsGaPClassesCreate)
 adminSettingsGaPClassesCreate =
     makeFrame PageAdminSettingsGaPClassesCreate
 
+adminSettingsGaPUserEdit :: ActionM r m => AUID User -> ServerT (FormHandler PageAdminSettingsGaPUsersEdit ST) m
+adminSettingsGaPUserEdit uid = redirectFormHandler editUserPage editUser
+  where
+    editUserPage = persistent $
+        PageAdminSettingsGaPUsersEdit
+        <$> ((\(Just u) -> u) <$> findUser uid) -- FIXME: Error handling
+        <*> getSchoolClasses
+
+    editUser (EditUser role clss) = do
+        let isSelectedClass (Student clss')    = clss == clss'
+            isSelectedClass (ClassGuest clss') = clss == clss'
+            isSelectedClass _                  = False
+
+        let addGroup gs = case role of
+                RoleStudent -> Student clss : gs
+                RoleGuest   -> ClassGuest clss : gs
+
+        persistent $ modifyUser uid (userGroups %~ addGroup . filter (not . isSelectedClass))
```

Comment:
This is confusing, could you write a comment what the application logic is that you had in mind?  I thought we'd probably want to just overwrite 'userGroups' and always keep a singleton in it.  Or, if that's not too much refactoring, change userGroups to `userGroup :: Group`.